spring-cloud-netflix-eureka-server/src/main/java/org/springframework/cloud/netflix/eureka/server/EurekaServerAutoConfiguration.java

```diff
@@ -207,8 +210,21 @@ public Jersey3DiscoveryClientOptionalArgs jersey3DiscoveryClientOptionalArgs() {
 
 	@Bean
 	public PeerAwareInstanceRegistry peerAwareInstanceRegistry(ServerCodecs serverCodecs,
-			EurekaServerHttpClientFactory eurekaServerHttpClientFactory) {
-		this.eurekaClient.getApplications(); // force initialization
+			EurekaServerHttpClientFactory eurekaServerHttpClientFactory,
+			EurekaInstanceConfigBean eurekaInstanceConfigBean) {
+		if (eurekaInstanceConfigBean.isAsyncClientInitialization()) {
+			log.debug("Initializing client asynchronously...");
+
+			ExecutorService executorService = Executors.newSingleThreadExecutor();
+			executorService.submit(() -> {
+				this.eurekaClient.getApplications();
+				log.debug("Asynchronous client initialization done.");
```

Comment:
Please check if debug logging is enabled (`log.isDebugEnabled()`)

Reply:
Done.